spec.bs

```diff
@@ -6,57 +6,1065 @@ Status: CG-DRAFT
+        1. Otherwise, resolve |promise| with undefined.
+    1. Return |promise|.
+  </div>
+
+  <div algorithm>
+    The <dfn method for="WorkletSharedStorage">clear()</dfn> method steps are:
+
+    1. Let |promise| be a new [=promise=].
+    1. Let |context| be {{WorkletSharedStorage}}'s {{SharedStorageWorkletGlobalScope}}'s [=outside settings=]'s [=target browsing context=].
+    1. [=Assert=] that |context| is not null.
+    1. Let |environment| be |context|'s [=active window=]'s [=relevant settings object=].
+    1. Let |queue| be |context|'s associated [=shared storage database/shared storage database queue=].
+    1. [=Enqueue the following steps=] on |queue|:
+        1. If the result of running [=shared storage database/clear all entries in the database=] with |queue| and |environment| is false, reject |promise| with a {{TypeError}}.
+        1. Otherwise, resolve |promise| with undefined.
+    1. Return |promise|.
+  </div>
+
+  ### Getter Methods ### {#getter}
+
+  <div algorithm>
+    The <dfn method for="WorkletSharedStorage">get(|key|)</dfn> method steps are:
+
+    1. Let |promise| be a new [=promise=].
+    1. Let |context| be {{WorkletSharedStorage}}'s {{SharedStorageWorkletGlobalScope}}'s [=outside settings=]'s [=target browsing context=].
+    1. [=Assert=] that |context| is not null.
+    1. If |key|'s [=string/length=] exceeds the [=key/maximum length=], return a [=promise rejected=] with a {{TypeError}}.
+    1. Let |environment| be |context|'s [=active window=]'s [=relevant settings object=].
+    1. Let |queue| be |context|'s associated [=shared storage database/shared storage database queue=].
+    1. [=Enqueue the following steps=] on |queue|:
+        1. Let |value| be the result of running [=shared storage database/retrieve an entry from the database=] with |queue|, |environment|, and |key|.
+        1. If |value| is null, reject |promise| with a {{TypeError}}.
+        1. Otherwise, if |value| is undefined, resolve |promise| with undefined.
+        1. Otherwise, resolve |promise| with |value|.
+    1. Return |promise|.
+  </div>
+
+  <div algorithm>
+    The <dfn method for="WorkletSharedStorage">length()</dfn> method steps are:
+
+    1. Let |promise| be a new [=promise=].
+    1. Let |context| be {{WorkletSharedStorage}}'s {{SharedStorageWorkletGlobalScope}}'s [=outside settings=]'s [=target browsing context=].
+    1. Let |environment| be |context|'s [=active window=]'s [=relevant settings object=].
+    1. Let |queue| be |context|'s associated [=shared storage database/shared storage database queue=].
+    1. [=Enqueue the following steps=] on |queue|:
+        1. Let |numEntries| be the result of running [=shared storage database/count entries in the database=] with |queue| and |environment|.
+        1. If |numEntries| is undefined, reject |promise| with a {{TypeError}}.
```

Comment:
Maybe another place you could use "is failure" instead.

Reply:
Done